tests/pyfunc/docker/test_docker_flavors.py

```diff
@@ -0,0 +1,407 @@
+from mlflow.models.flavor_backend_registry import get_flavor_backend
+
+from tests.catboost.test_catboost_model_export import reg_model  # noqa: F401
+from tests.diviner.test_diviner_model_export import (  # noqa: F401
+    diviner_data,
+    diviner_groups,
+    grouped_prophet,
+)
+from tests.fastai.test_fastai_model_export import fastai_model as fastai_model_raw  # noqa: F401
+from tests.h2o.test_h2o_model_export import h2o_iris_model  # noqa: F401
+from tests.helper_functions import get_safe_port
+from tests.langchain.test_langchain_model_export import fake_chat_model  # noqa: F401
+from tests.lightgbm.test_lightgbm_model_export import lgb_model  # noqa: F401
+from tests.models.test_model import iris_data, sklearn_knn_model  # noqa: F401
+from tests.paddle.test_paddle_model_export import pd_model  # noqa: F401
+from tests.pmdarima.test_pmdarima_model_export import (  # noqa: F401
+    auto_arima_object_model,
+    test_data,
+)
+from tests.prophet.test_prophet_model_export import prophet_model as prophet_raw_model  # noqa: F401
+from tests.pyfunc.docker.conftest import (  # noqa: F401
+    MLFLOW_ROOT,
+    TEST_IMAGE_NAME,
+    docker_client,
+    save_model_with_latest_mlflow_version,
+)
+from tests.spacy.test_spacy_model_export import spacy_model_with_data  # noqa: F401
+from tests.spark.test_spark_model_export import (  # noqa: F401
+    iris_df,
+    spark,
+    spark_model_iris,
+)
+from tests.statsmodels.model_fixtures import ols_model
+from tests.tensorflow.test_tensorflow2_core_model_export import tf2_toy_model  # noqa: F401
+from tests.transformers.helper import load_small_seq2seq_pipeline
+
+
+@pytest.mark.skipif(os.getenv("GITHUB_ACTIONS") == "true", reason="Time consuming tests")
+@pytest.mark.parametrize(
+    ("flavor"),
+    [
+        "catboost",
+        "diviner",
+        "fastai",
+        "h2o",
+        # "johnsnowlabs", # Couldn't test JohnSnowLab locally due to license issue
+        "keras",
+        "langchain",
+        "lightgbm",
+        # "mleap", # Mleap model logging is deprecated since 2.6.1
+        "onnx",
+        # "openai", # OPENAI API KEY is not necessarily available for everyone
+        "paddle",
+        "pmdarima",
+        "prophet",
+        "pyfunc",
+        "pytorch",
+        "sklearn",
+        "spacy",
+        "spark",
+        "statsmodels",
+        "tensorflow",
+        "transformers",
+    ],
+)
+def test_build_image_and_serve(flavor, request):
+    model_path = request.getfixturevalue(f"{flavor}_model")
+
+    # Build an image
+    backend = get_flavor_backend(model_uri=model_path, docker_build=True)
+    backend.build_image(
+        model_uri=model_path,
+        image_name=TEST_IMAGE_NAME,
+        mlflow_home=MLFLOW_ROOT,  # Required to prevent installing dev version of MLflow from PyPI
+    )
+
+    # Run a container
+    port = get_safe_port()
+    docker_client.containers.run(
+        image=TEST_IMAGE_NAME,
+        ports={8080: port},
+        detach=True,
+    )
```

Comment:
Can we kill this container when it's no longer needed?

Reply:
Could you elaborate a bit? They are cleaned up in fixture at the end of each test case now.

Reply:
Ah I see, `clean_up_docker_image` is in `tests/pyfunc/docker/conftest.py`.